libfwupdplugin/fu-volume.c

```diff
@@ -1087,11 +1087,15 @@ fu_volume_new_by_kind(const gchar *kind, GError **error)
 		/* ignore a partition that claims to be a recovery partition */
 		if (g_strcmp0(kind, FU_VOLUME_KIND_BDP) == 0 ||
 		    g_strcmp0(kind, FU_VOLUME_KIND_ESP) == 0) {
-			g_autofree gchar *name = fu_volume_get_partition_name(vol);
-
-			if (name == NULL)
-				name = fu_volume_get_block_name(vol);
-			if (name != NULL) {
+			g_autofree gchar *partition = fu_volume_get_partition_name(vol);
+			g_autofree gchar *block = fu_volume_get_block_name(vol);
+			const gchar *name;
+
+			if (partition != NULL && strlen(partition) > 0)
```

Comment:
rather than doing this, maybe we should not set the zero length string -- I'd much prefer something like https://github.com/fwupd/fwupd/pull/7632